packages/react-rest/src/services/react-hooks.ts

```diff
@@ -1,21 +1,78 @@
 import { useState, useEffect } from "react"
-import { getList, subscribeToList } from "data-core"
-import type { Source, Record, QueryList } from "data-core"
+import { createOne, getList, subscribeToList } from "data-core"
+import type { CreateData, Meta, Source, Record, QueryList } from "data-core"
+import { getRecords } from "data-core"
 
+/**
+ * Fetches a list of records using the data-core getList function,
+ * subscribes to the store for updates to the list, returns the list.
+ */
 export const useList = (
   dataSource: Source,
   schema: string,
   query: QueryList,
-): [Record[]] => {
-  const [data, setData] = useState<Record[]>([]) // @todo default to what's in the store
+): [Record[], Meta] => {
+  const defaultData = getRecords(schema)
+  const [data, setData] = useState<Record[]>(defaultData)
+  const [error, setError] = useState<Error | undefined>(undefined)
+  const [loading, setLoading] = useState<boolean>(true)
 
   useEffect(() => {
-    getList(dataSource, schema, query).then(setData)
+    setLoading(true)
+    getList(dataSource, schema, query)
+      .then(setData)
+      .catch(setError)
+      .finally(() => setLoading(false))
   }, [dataSource, schema, query])
 
   useEffect(() => {
     return subscribeToList(schema, (records: Record[]) => setData(records))
   }, [schema])
 
-  return [data]
+  const status = (
+    error ? "error" : loading ? "loading" : "success"
+  ) as Meta["status"]
+  const meta = {
+    status,
+    error,
+    isLoading: status === "loading",
+    isDone: status === "success",
+    isRejected: status === "error",
+  }
+
+  return [data, meta]
+}
+
+/**
+ * Returns a function that creates a new record using the data-core createOne,
+ * @todo metadata, and the last created record.
+ */
+export const useCreateOne = (
+  dataSource: Source,
+  schema: string,
+): [(data: CreateData) => void, Meta, Record?] => {
+  const [data, setData] = useState<Record | undefined>(undefined)
+  const [error, setError] = useState<Error | undefined>(undefined)
+  const [loading, setLoading] = useState<boolean>(false)
+
+  function create(data: CreateData) {
+    setLoading(true)
+    createOne(dataSource, schema, data)
+      .then(setData)
+      .catch(setError)
+      .finally(() => setLoading(false))
+  }
+
+  const status = (
+    error ? "error" : loading ? "loading" : "success"
+  ) as Meta["status"]
```

Comment:
was getting typescript errors without this assertion, not sure why it wasn't working without 😢

Reply:
little internal things like this can be adjusted as we go. As long as the users don't need `as`, I'm happy now.

Reply:
gonna leave this one as is til later teardown review. it doesn't like when I wrap the entire ternary and cast `as const`, so leaving `as Meta["status"]` rather than casting each string on line 67 as a const